### PR TITLE
[Merged by Bors] - feat(Mathlib/Tactic/Replace): add version of `replace` without `:=`

### PR DESCRIPTION
**Pull request by dupuisf**
As @digama0 said in #332, `replace` was missing part of its implementation, namely the case where the proof is not immediately given but rather deferred to an additional goal. This PR adds this case.

### CODE REVIEW

**Comment by dupuisf** [comment]
Here the linter is complaining about the docstring for `replace'`, but I have no idea where that declaration is or how to add a docstring to it.

**Comment by digama0** [comment]
`replace'` is the syntax declaration on line 14. That's where you should put front end documentation for the tactic, since that's what users see on a hover.

**Comment by dupuisf** [comment]
> `replace'` is the syntax declaration on line 14. That's where you should put front end documentation for the tactic, since that's what users see on a hover.

That's what I figured, but adding a docstring above that line does not satisfy the linter.

File: Mathlib/Tactic/Replace.lean [hunk 1]
```diff
@@ -53,3 +55,47 @@ elab_rules : tactic
         try replaceMainGoal [← Meta.clear (← getMainGoal) hId]
         catch | _ => pure ()
       | none     => pure ()
+
+/--
```
Comment (digama0) [inline Mathlib/Tactic/Replace.lean:59]: This docstring shoul move to above the `syntax` line

File: Mathlib/Tactic/Replace.lean [hunk 2]
```diff
@@ -11,7 +11,6 @@ namespace Mathlib.Tactic
 open Lean Elab.Tactic
 
 syntax "replace " haveDecl : tactic
```
Comment (digama0) [inline Mathlib/Tactic/Replace.lean:13]: I realize it's not actually added in this PR, but could you also add a docstring or copy paste the one from `replace'` here?

Comment (dupuisf) [reply Mathlib/Tactic/Replace.lean]: Sure -- done!

File: Mathlib/Tactic/Replace.lean [hunk 3]
```diff
@@ -53,3 +54,49 @@ elab_rules : tactic
...
+```lean
+case h
+f: α → β
+h: α
+⊢ β
+
+f: α → β
+h✝: α
+h: β
+⊢ goal
```
Comment (semorrison) [inline Mathlib/Tactic/Replace.lean:86]:
```suggestion
f : α → β
h : β
⊢ goal
```

whereas `have h : β` would result in:

```lean
case h
f : α → β
h : α
⊢ β

f : α → β
h✝ : α
h : β
⊢ goal
```

**Comment by semorrison** [comment]
Let's also add
```
-- tests without `:=`, creating a new subgoal

example : Nat := by
  have h : Int := 0
  replace h : Nat
  exact 0
  assumption
```
or similar in `test/Replace.lean`.

**Comment by digama0** [comment]
bors merge

**Pull request closed without merging**